classes/tar.py

```diff
@@ -1,31 +1,75 @@
-import tarfile, os
-from typing import Union
+import tarfile
+import os
+from typing import Union, List, Dict
 
 class Tar:
     def __init__(self):
         pass
-    
-    def compress(self, source: Union[str, list, dict], output: str, exclude_paths = []) -> str:
 
+    def compress(self, source: Union[str, List[str], Dict], output: str, symlink: bool, exclude_paths: List[str] = []) -> str:
+        
         def exclude_path(tarinfo, folder_name):
             file_name = tarinfo.name.replace(folder_name, "")
             for exclude_path in exclude_paths:
                 if file_name.startswith(exclude_path):
-                    print(f"Excluding {tarinfo.name}")
                     return None
             return tarinfo
-    
-        with tarfile.open(output, "w:gz") as tar:
-            if type(source) != str:
-                for path in source:
-                    # remove last folder to get base path
-                    folder_name = path.split("/")[-1] + "/"
-
-                    if not os.path.exists(path):
-                        print(f"Skipped, {path} not found")
-                        continue
-                    tar.add(path, arcname=os.path.basename(path), filter=lambda tarinfo: exclude_path(tarinfo, folder_name))
-            else:
-                tar.add(path, arcname=os.path.basename(path))
-            tar.close()
-        return output
\ No newline at end of file
+
+        if isinstance(source, list):
+            base_name = os.path.basename(source[0])
+        else:
+            base_name = os.path.basename(source)
+
+        if os.path.isdir(output):
+            output_path = os.path.join(output, f"{base_name}.tar.gz")
+        else:
+            output_path = output
+
+        if not symlink:
+            with tarfile.open(output_path, "w:gz") as tar:
+                if isinstance(source, list):
+                    for path in source:
+                        if not os.path.exists(path):
+                            continue
+
+                        for root, dirs, files in os.walk(path):
+                            if os.path.islink(root):
+                                continue
+                            
+                            for file in files:
+                                full_path = os.path.join(root, file)
+                                if os.path.islink(full_path):
+                                    continue
+                                
+                                arcname = os.path.join(base_name, os.path.relpath(full_path, path))
+                                tar.add(full_path, arcname=arcname, filter=lambda tarinfo: exclude_path(tarinfo, base_name))
```

Comment:
kenapa gak `dereference=symlink` aja, jadi kalo `not symlink` dereference nya `False` artinya symlink ikut terbackup cuman isi didalam nya nggak, sebaliknya kalo dereference nya `True` symlink terbackup berserta isi dalamnya.

Reply:
Bisa aja kak, Meskipun opsi dereference=False dipilih symlink atau file link tetap akan ikut terkompres : 

![image](https://github.com/user-attachments/assets/a4624650-049e-4597-937c-998354de87ee)